tests/checks/test_pkgdir.py

```diff
@@ -279,13 +279,79 @@ def test_multiple_files_over_limit(self):
                 f.seek(size)
                 f.write('\0')
         r = self.assertReports(self.mk_check(), [pkg])
-        assert len(r) == 2
+        assert len(r) == 3
         assert isinstance(r[0], pkgdir.SizeViolation)
         assert isinstance(r[1], pkgdir.SizeViolation)
+        assert isinstance(r[2], pkgdir.TotalSizeViolation)
         assert (
-            tuple(sorted((x.filename, x.size) for x in r)) ==
+            tuple(sorted((x.filename, x.size) for x in r[:2])) ==
             (('files/massive', 1024*100+1), ('files/over', 1024*20+1))
         )
+        assert r[2].size == 1024*(10+20+20+100)+4-1
+
+
+class TestManifestSizeViolation(PkgDirCheckBase):
+    """Check ManifestSizeViolation results."""
+
+    def write_manifest(self, pkg, size):
+        with open(pjoin(os.path.dirname(pkg.path), 'Manifest'), 'w') as f:
+            f.seek(size - 1)
+            f.write('\0')
```

Comment:
I'm not that sure about this code. My knowledge about filesystems is lacking in this regard, and I'm unsure if this will always work on all filesystems.
Maybe add small `assert` in this `write_manifest`that check the size of this file matches (and then add good assert message that the test setup is wrong on this FS and don't blame the package).